include/igl/readOBJ.cpp

```diff
@@ -77,22 +79,18 @@ IGL_INLINE bool igl::readOBJ(
       char * l = &line[strlen(type)];
       if(type == v)
       {
-        double x[4];
-        int count =
-        sscanf(l,"%lf %lf %lf %lf\n",&x[0],&x[1],&x[2],&x[3]);
-        if(count != 3 && count != 4)
+        std::istringstream ls(&line[1]);
+        std::vector<Scalar > vertex{ std::istream_iterator<Scalar >(ls), std::istream_iterator<Scalar >() };
+
+        if (vertex.size() < 3)
```

Comment:
I guess if we're going to allow n>3 why not also allow n==2 (or I suppose n==1?)?  Do people use .obj to store 2D meshes? 

_I'm guessing most people are not really storing 4+d meshes but rather sneaking per-vertex data into the format, so maybe this is all in the name of hackery_.